components/scream/src/diagnostics/tests/field_at_single_pressure_tests.cpp

```diff
@@ -0,0 +1,166 @@
+#include "catch2/catch.hpp"
+
+#include "ekat/ekat_pack_utils.hpp"
+
+#include "diagnostics/field_at_single_pressure.hpp"
+
+#include "share/grid/mesh_free_grids_manager.hpp"
+#include "share/field/field_utils.hpp"
+#include "share/util/scream_setup_random_test.hpp"
+
+namespace scream {
+
+std::shared_ptr<GridsManager>
+create_gm (const ekat::Comm& comm, const int ncols, const int nlevs) {
+
+  const int num_global_cols = ncols*comm.size();
+
+  ekat::ParameterList gm_params;
+  gm_params.set<int>("number_of_global_columns", num_global_cols);
+  gm_params.set<int>("number_of_vertical_levels", nlevs);
+
+  auto gm = create_mesh_free_grids_manager(comm,gm_params);
+  gm->build_grids();
+
+  return gm;
+}
+
+TEST_CASE("field_at_single_pressure")
+{
+
+  // Test that output at a single pressure level works as expected.
+  // For this test we set a field "M" to be defined as 100*i + k,
+  // where i=column and k=level
+  //
+  // We then set the pressure levels to be 100*k where again k=level.
+  //
+  // Lastly we define a pressure_at_single_level to be for p=150, thus
+  // the output should be (M_1+M_2)/2, or halfway between level 1 and level
+  // 2 of the data.  Given the formula above the output should be exactly
```

Comment:
I would prefer a bunch of tests covering multiple cases, rather than a single one duplicated over cols/levs. You can keep the pressure levels as you did, but you should test the diagnostic at left/right ends (p=100, p=200), middle (p=150), and also random entry (which might require to use `Catch::Approx` to account for roundings).